xtensa-lx106/Dockerfile

```diff
@@ -0,0 +1,60 @@
+FROM dockcross/base
+MAINTAINER Roman Valls Guimera "brainstorm@nopcode.org"
+
+# For embedded IoT systems such as the ESP82xx from Espressif
+
+RUN echo "deb http://ftp.de.debian.org/debian jessie main non-free" > /etc/apt/sources.list.d/non-free.list \
+&& apt-get update && apt-get install -y \
+  qemu-user \
+  qemu-user-static \
+	autoconf \
+	automake \
+	bison \
+	bzip2 \
+	flex \
+	g++ \
+	gawk \
+	gcc \
+	git \
+	gperf \
+	libexpat-dev \
+	libtool \
+	libtool-bin \
+	make \
+	ncurses-dev \
+	nano \
+	python \
+	python-dev \
+	python-serial \
+	sed \
+	texinfo \
+	rar \
+	unzip \
+	wget \
+	help2man \
+	git \
+&& apt-get clean
+
+RUN useradd -ms /bin/bash espbuild
+USER espbuild
```

Comment:
We will want to set the `USER` back to `root` after the build has completed for the dockcross scripts to work correctly. We may want to just build as root unless there is a good reason not to to avoid conflicts with the UID of the dockcross invoking user.

Reply:
Well, [not in this case, @thewtex](https://circleci.com/gh/brainstorm/cross-compilers/68):

```
Now configured for "xtensa-lx106-elf"
make[2]: Leaving directory '/home/espbuild/esp-open-sdk/crosstool-NG'
sed -r -i.org s%CT_PREFIX_DIR=.*%CT_PREFIX_DIR="/home/espbuild/esp-open-sdk/xtensa-lx106-elf"% .config
sed -r -i s%CT_INSTALL_DIR_RO=y%"#"CT_INSTALL_DIR_RO=y% .config
cat ../crosstool-config-overrides >> .config
./ct-ng build
make[2]: Entering directory '/home/espbuild/esp-open-sdk/crosstool-NG'

[ERROR]  You must NOT be root to run crosstool-NG
```

I guess that PR #152 could potentially help here, but for now we have to use regular user unless we want to patch crosstool-NG...

Reply:
Yes, we should use the new dockcross crosstool-ng infrastructure.